web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -861,6 +857,48 @@ public ResponseEntity<List<GenomicDataBin>> fetchGenomicDataBinCounts(
         return new ResponseEntity<>(studyViewFilterApplier.getDataBins(dataBinMethod, interceptedGenomicDataBinCountFilter), HttpStatus.OK);
     }
 
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', T(org.cbioportal.utils.security.AccessLevel).READ)")
+    @RequestMapping(value = "/genomic-data-counts/fetch", method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch genomic data counts by GenomicDataCountFilter")
+    public ResponseEntity<List<GenomicDataCount>> fetchGenomicDataCounts(
+        @ApiParam(required = true, value = "Genomic data bin count filter") @Valid @RequestBody(required = false) GenomicDataCountFilter genomicDataCountFilter,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
+        @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
+        @ApiParam(required = true, value = "List of copy number count identifiers")
+        @ApiIgnore
+        @Valid @RequestAttribute(required = false, value = "interceptedGenomicDataCountFilter") GenomicDataCountFilter interceptedGenomicDataCountFilter
+    ) throws StudyNotFoundException {
+        List<GenomicDataFilter> gdFilters = interceptedGenomicDataCountFilter.getGenomicDataFilters();
+        StudyViewFilter studyViewFilter = interceptedGenomicDataCountFilter.getStudyViewFilter();
+        // when there is only one filter, it means study view is doing a single chart filter operation
+        // remove filter from studyViewFilter to return all data counts
+        // the reason we do this is to make sure after chart get filtered, user can still see unselected portion of the chart
+        if (gdFilters.size() == 1) {
+            studyViewFilterUtil.removeSelfFromGenomicDataFilter(
+                gdFilters.get(0).getHugoGeneSymbol(), 
+                gdFilters.get(0).getProfileType(), 
+                studyViewFilter);
+        }
+        List<SampleIdentifier> filteredSampleIdentifiers = studyViewFilterApplier.apply(studyViewFilter);
+
+        if (filteredSampleIdentifiers.isEmpty()) {
+            return new ResponseEntity<>(new ArrayList<>(), HttpStatus.OK);
+        }
+
+        List<String> studyIds = new ArrayList<>();
+        List<String> sampleIds = new ArrayList<>();
+        studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, studyIds, sampleIds);
+        
+        List<GenomicDataCount> result = studyViewService.getCNAAlterationCountsByGeneSpecific(
+            studyIds,
+            sampleIds,
+            gdFilters.stream().map(GenomicDataFilter::getHugoGeneSymbol).collect(Collectors.toList()),
+            studyViewFilter.getAlterationFilter());
```

Comment:
`studyViewFilter` is used in filtering data. This shouldn't be used here.

Reply:
What is the alterationFilter to be used? Do you mean to create a brand new alterationFilter in StudyViewService?